python/cudf_polars/cudf_polars/dsl/expressions/datetime.py

```diff
@@ -125,6 +126,11 @@ def __init__(
         if self.name not in self._valid_ops:
             raise NotImplementedError(f"Temporal function {self.name}")
 
+        if self.name is TemporalFunction.Name.ToString and plc.traits.is_duration(
```

Comment:
Actually can we put this check in `do_evaluate` so we only have to check `self.name is TemporalFunction.Name.ToString` once? e.g.

```python
if self.name is TemporalFunction.Name.ToString:
    if plc.traits.is_timestamp(column.obj.type()):
         return ...
    else:
         raise NotImplementedError(...)
```

Reply:
No, because then we wouldn't fall back during translation for unsupported expressions, I think